x-pack/plugins/infra/server/lib/alerting/metric_threshold/lib/evaluate_rule.ts

```diff
@@ -202,20 +203,23 @@ const getMetric: (
         bucketSelector,
         afterKeyHandler
       )) as Array<Aggregation & { key: Record<string, string>; doc_count: number }>;
-      const groupedResults = compositeBuckets.reduce(
-        (result, bucket) => ({
-          ...result,
-          [Object.values(bucket.key)
-            .map((value) => value)
-            .join(', ')]: getValuesFromAggregations(
-            bucket,
-            aggType,
-            dropPartialBucketsOptions,
-            calculatedTimerange,
-            bucket.doc_count
-          ),
-        }),
-        {}
+      const groupedResults: Record<string, any> = {};
+      await Promise.all(
```

Comment:
I'm not sure I get the benefits of running this computation asynchronously. We register all the `setTimeout` callbacks in the same tick with the same delay so they should end up being queued and executed serially in a single tick once the event loop reaches the `timers` phase, which gives us the same execution model as a synchronous loop but at a later time

Reply:
I switched this to a simple for/of loop and it's just as performant. Good catch.